File: app/src/main/java/com/ntech/fourtop/ui/MainActivity.java
package com.ntech.fourtop.ui;

import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.navigation.NavController;
import androidx.navigation.NavDestination;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import com.ntech.fourtop.R;
import com.ntech.fourtop.databinding.ActivityMainBinding;

import static androidx.navigation.Navigation.findNavController;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    private NavController navController;
    private AppBarConfiguration appBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        setSupportActionBar(binding.toolbar);

        navController = findNavController(this, R.id.main_nav_host);
        appBar = new AppBarConfiguration.Builder(navController.getGraph()).build();
        NavigationUI.setupWithNavController(binding.toolbar, navController, appBar);

        binding.navigationView.setNavigationItemSelectedListener(this::navigationListener);
        navController.addOnDestinationChangedListener(this::destinationChanged);

        switch (navController.getCurrentDestination().getId()) {
            case R.id.logoFragment:
            case R.id.loginFragment:
            case R.id.registerFragment:
                binding.toolbar.setVisibility(View.GONE);
                break;
        }
    }

    @Override
    public boolean onSupportNavigateUp() {
        return NavigationUI.navigateUp(navController, appBar) || super.onSupportNavigateUp();
    }

    @Override
    public void onBackPressed() {
        int currentFragment = navController.getCurrentDestination().getId();

        switch (currentFragment) {
            case R.id.loginFragment:
            case R.id.homeFragment:
                finish();
                break;
            default:
                navController.popBackStack();
                break;
        }
    }

    private boolean navigationListener(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.menu_nav_profile:
                navController.navigate(R.id.action_homeFragment_to_profileFragment);
                break;
            case R.id.menu_nav_user_images:
                navController.navigate(R.id.action_homeFragment_to_userImagesFragment);
                break;
            case R.id.menu_nav_voting:
                navController.navigate(R.id.action_homeFragment_to_votingFragment);
                break;
            case R.id.menu_nav_top_images:
                navController.navigate(R.id.action_homeFragment_to_topImagesFragment);
                break;
            case R.id.menu_nav_top_users:
                navController.navigate(R.id.action_homeFragment_to_topUsersFragment);
                break;
        }

        NavigationUI.onNavDestinationSelected(item, navController);
        binding.layoutDrawer.closeDrawers();
        return true;
    }

    private void destinationChanged(@NonNull NavController controller,
                                    @NonNull NavDestination destination,
                                    @NonNull Bundle arguments) {
        if (destination.getId() == R.id.homeFragment) {
            binding.layoutDrawer.setDrawerLockMode(DrawerLayout.LOCK_MODE_UNLOCKED);
        } else {
            binding.layoutDrawer.setDrawerLockMode(DrawerLayout.LOCK_MODE_LOCKED_CLOSED);
        }
    }
}

File: app/src/main/java/com/emikhalets/voteapp/view/MainActivity.kt
package com.emikhalets.voteapp.view

import android.os.Bundle
import androidx.activity.result.contract.ActivityResultContracts.RequestPermission
import androidx.appcompat.widget.Toolbar
import androidx.navigation.NavController
import androidx.navigation.fragment.NavHostFragment
import com.emikhalets.voteapp.R
import com.emikhalets.voteapp.databinding.ActivityMainBinding
import com.emikhalets.voteapp.utils.CAMERA
import dagger.android.AndroidInjection
import dagger.android.support.DaggerAppCompatActivity

class MainActivity : DaggerAppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var navController: NavController

    lateinit var drawer: AppDrawer
    lateinit var toolbar: Toolbar

    private val permissionResult = registerForActivityResult(RequestPermission()) {}

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        AndroidInjection.inject(this)
        initNavigation()
        initViews()
        permissionResult.launch(CAMERA)
    }

    override fun onBackPressed() {
        when (navController.currentDestination?.id) {
            R.id.authLoginFragment -> finish()
            R.id.homeFragment -> finish()
            else -> super.onBackPressed()
        }
    }

    private fun initNavigation() {
        val navHostFragment =
                supportFragmentManager.findFragmentById(R.id.fragment_container) as NavHostFragment
        navController = navHostFragment.navController
    }

    private fun initViews() {
        toolbar = binding.toolbar
        setSupportActionBar(toolbar)
        drawer = AppDrawer(this)
        drawer.create()
    }
}